core/src/main/scala/org/apache/spark/status/api/v1/ApplicationListResource.scala

```diff
@@ -40,9 +40,11 @@ private[v1] class ApplicationListResource(uiRoot: UIRoot) {
         status
       }
     }
+
+    val numApps = Option(limit).getOrElse(Integer.MAX_VALUE).asInstanceOf[Int]
```

Comment:
Yup the iterator works fine if I limit the number of applications. I just updated this line in the PR for archive use.